llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -822,12 +822,47 @@ template <typename Class> struct bind_ty {
   }
 };
 
+template <typename Class, typename MatchTy> struct bind_and_match_ty {
```

Comment:
would it be possible to just use m_CombineAnd to make the new matchers? Not sure if that would save much though

Reply:
Sounds good to leave it as is, perhaps with a brief comment.